### DISCUSSION TITLE
TypeScript/JSDoc for inline documentation

### DISCUSSION THREAD

**Discussion by jsjoeio** [johnlindquist/kit#61] [Ideas]
It would be nice if I could get some documentation in my editor about some of the utilities that Kit provides out of the box. 

<img width="1904" alt="image" src="https://user-images.githubusercontent.com/3806031/111709563-0f6dcc00-8805-11eb-8d22-2d204d504e5f.png">

Maybe this could be achieved with TypeScript or JSDoc?

**Comment by johnlindquist** [comment]
Yah, the current plan is a `kit.d.ts` at the root of the .kenv dir that declares all the global functions.

I've tried the jsdoc inline, but it fails to pick it up since we're not using standard import syntax.

**Reply by jsjoeio** [reply]
Sweet! Glad to hear that's on the roadmap :)
